### PR TITLE
Rewrite ObjPylon01.c in C++

### PR DESCRIPTION
**Pull request by riidefi**

### CODE REVIEW

**Comment by riidefi** [comment]
Only done to allow `RaceConfig.h` to be deleted. A follow-up PR could include a proper rewrite

File: payload/game/obj/ObjPylon01.cc [hunk 1]
```diff
@@ -0,0 +1,16 @@
+extern "C" {
+#include "ObjPylon01.h"
+}
+
+#include "../system/RaceConfig.hh"
```
Comment (vabold) [inline payload/game/obj/ObjPylon01.cc:5]:
```suggestion
#include "game/system/RaceConfig.hh"
```

File: payload/game/obj/ObjPylon01.cc [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+extern "C" u32 my_ObjPylon01_vf_b0(const ObjPylon01 *self) {
+    auto *cfg = System::RaceConfig::Instance();
+    if (cfg->raceScenario().players[0].type != System::RaceConfig::Player::Type::Ghost &&
+            self->playerId != 0) {
+        return 1; // Transparent
+    }
+
+    return 12; // Opaque
+}
+PATCH_B(ObjPylon01_vf_b0, my_ObjPylon01_vf_b0);
```
Comment (vabold) [inline payload/game/obj/ObjPylon01.cc:16]: Please use the `REPLACE` macro.

**Review by vabold — CHANGES_REQUESTED** [review]

File: payload/game/obj/ObjPylon01.cc (hunk 1 above)
Comment (riidefi) [reply payload/game/obj/ObjPylon01.cc:5]: It was this way in the original file.

File: payload/game/obj/ObjPylon01.cc (hunk 2 above)
Comment (riidefi) [reply payload/game/obj/ObjPylon01.cc:16]: It was this way in the original file.

**Comment by riidefi** [comment]
Closed in favor of [#820](https://github.com/camzillajs101/garfield-cursed-image/blob/master/app/assets/images/amalgam.jpg?raw=true)

**Pull request closed without merging**